### ISSUE TITLE
Auto start form rc.local

### ISSUE THREAD

**Issue by dmking043** [shrocky2/SiriGarage#7]
Hello, great project, I am not that advanced, but working my way through. I added the 2 lines to the /etc/rc.local file, but when I reboot, the server is not running in the background as you indicated. Not sure what I am doing wrong, I have the lines exactly as you have them right before the 'exit 0' an this is my folder where the app is located as well.

sudo python3 /home/pi/Downloads/SiriGarage/siriweb.py &
sudo python3 /home/pi/Downloads/SiriGarage/log.py &

exit  0

Please let me know your thoughts.

One last question is there any way to mask the Password on the setting page, when I type in the password it is visible, not sure if I am doing something wrong, missed a setting or if that is expected.

Thanks for your help

**Comment by shrocky2** [comment]
try running the program from the command line "sudo python3 /home/pi/Downloads/SiriGarage/siriweb.py" and see if the program starts up or give you errors. If you get errors, then that would explain why the rc.local file doesn't start up the program. Also do the same for the log file and check for errors too "sudo python3 /home/pi/Downloads/SiriGarage/log.py"

As for the password on the settings page, it will not hide the password because I didn't program it that way. You can edit the file yourself and make it so it doesn't show the password. Open the "settings.txt" file located in the "templates" folder and edit lines 67 & 77...specifically the part that says (input type="text")...change that to (input type="password")

Good luck and thanks for checking out my project.

**Comment by shrocky2** [comment]
I see that you are running the program from withing a Downloads folder. If you keep getting errors, try to move the entire project out of the Downloads folder and into the root folder (/home/pi/)...this project might be having trouble locating files. Take a look at this issue another user submitted and my first response seemed to fix their issue. https://github.com/shrocky2/SiriGarage/issues/6

**Comment by dmking043** [comment]
Thanks for the quick reply, I tried your suggestions on running the app, all is good running manually. I read several articles on rc.local vs. crontab. I created a script with the 2 lines, I ran it in crontab, I also followed a suggestion to pause the start of the script for 30 seconds allowing the startup to complete before attempting to run the app, "@reboot /bin/sleep 30;"  it works as expected, no issues. So I think I am good. 

I will make the change on the password you suggested, I appreciate that.

I also found a great case for your project on Thingiverse, I have a 3d printer so I am printing it out, https://www.thingiverse.com/thing:2975944, see attached pictures.  Thought you might be interested. 

Thanks again for the project, great job!

![Rpi 4C Relay Case2022](https://user-images.githubusercontent.com/45828319/210083869-c3de15cc-6f4e-4dcc-bdf2-605284104872.jpg)
![Rpi 4C Relay Case a 2022](https://user-images.githubusercontent.com/45828319/210084176-4676c7b5-d2fd-4300-b0bf-9f04289f49ec.jpg)

**Comment by dmking043** [comment]
Sorry, I do need some additional help when you have time, When setting up Siri Shortcuts, I am able to setup the Status, that works as expected. But when trying to setup the Open command, following your instructions is not working on Step 28. When I search for Speak Text. It doesn't default to "Speak Get contents of URL" It defaults to "Speak TEXT" and I can't seem to change it. I have the latest IOS 16.2 and it doesn't exactly follow your screenshots as you must have done this on a earlier version of IOS. I can't get past this, it just doesn't work. I will keep digging around, but if you have any suggestions, I would greatly appreciate it Thanks

**Comment by shrocky2** [comment]
I'm having trouble trying to recreating your problem. I would recommend starting fresh with a new shortcut so hopefully the 2 command get link together. On the Speak command, you should be able to click on "Text" and below "Select Variable" should show up, right below the "Get Contents of" field a little "Contents of URL" hopefully shows up. Hope this works for you.
![IMG_9272](https://user-images.githubusercontent.com/33706433/210160648-9c80276c-1c97-4f83-a06d-73b90ee950e7.jpg)
![IMG_9271](https://user-images.githubusercontent.com/33706433/210160649-9e43d443-f2fe-4a06-9772-c6e357c5ec1a.jpg)

**Comment by dmking043** [comment]
Happy New Years!!! Thanks for responding, I tried many times and I think I
have it setup correctly, see screenshots attached, but all Siri responds is
"We have a problem" nothing more.

I will keep at it, sorry for bothering you, I figured I would give it one
last try, not sure what I am doing wrong, but I have started over and
followed the steps exactly, at least I believe I have. The Door Status
works, Siri can respond with the correct Door Status, I will try to open
the door with the passcode once I figure out how to do that to see if that
works, but if you have any suggestions, I would appreciate it. Thanks again
for taking the time to help me.

Appreciated.

**Comment by dmking043** [comment]
Steve, maybe the issue is that trying to manually open the garage door with
the "Click to operate garage door" and enter the default PW: 12345678, it
does nothing.

I did the relaytest and the test works, it cycles through each relay one at
a time and continues until I hit Ctrl + C, so maybe the Shortcut is set up
correctly for Siri but the actual process to initiate the relay is where
the issue is. Does the Relay need to be connected to the Garage Door Opener
to test? I will try that next as I am just testing  and haven't
hooked everything up just yet.

Please let me know your thoughts, Thanks

**Comment by shrocky2** [comment]
No, the relay doesn't need to be connected to the garage door in order to test. You will still see a single relay trigger, just like it did when you did the relay test. Did you "Enable Siri" within the settings. I believe the default is Disabled

**Comment by dmking043** [comment]
I doubled checked. It is set to “yes”. Even if I control the door and enter the default password which it is set to the relay never triggers. My only other thought it to start from scratch. Maybe the code got corrupted possibly? Even if I still don’t have Siri properly setup  I don’t see any reason for the relay not to trigger when entering the password, especially since the relay test works. I must be doing something wrong. I will start over and let you know.  Thanks for your help. Again sorry to be a pain.  Sent from my iPhoneOn Jan 1, 2023, at 12:23 PM, Steve ***@***.***> wrote:﻿
No, the relay doesn't need to be connected to the garage door in order to test. You will still see a single relay trigger, just like it did when you did the relay test. Did you "Enable Siri" within the settings. I believe the default is Disabled

**Comment by dmking043** [comment]
Steve, just wanted to let you know I started from scratch. I reloaded the
Pi with the latest Raspberry Pi OS, I then followed your instructions step
by step, and  Voila!!! It is Working!!!

I clearly must have corrupted the code or there was something wrong with
the Pi I used as it was used for another project and possibly I
just corrupted something.

I will make the default changes as you mentioned one at a time to make sure
it stays working, but I wanted to let you know and I really appreciate your
time, support and help,
